### PR TITLE
fix: conditional CSS in Vue templates, close #7

### LINKED ISSUE #7: Conditional classes not being generated
**Problem**
It seems this plugin isn't generating classes for conditional classes in vue templates. For example the following worked when using tailwindcss, but it no longer works with windicss

```html
<div :class="{ 'ml-24': hasSidebar }"></div>
```
If I move it to just a normal class attribute then it works, but obviously no longer produces the desired effet.
```html
<div class="ml-24"></div>
```

Though I'm not sure if this is a problem with windicss or the plugin

### PR DESCRIPTION
**Pull request by ElMassimo**
### Description 📖 

This pull request fixes the scanner so that it can parse conditional CSS rules in Vue templates.

Checked on the example and in one of my projects, but since regexes are tricky and this one has had many iterations, would like @antfu to review it before merging.

### Notes ✏️ 

Using ```[^"'`<>,]``` is a simple way to avoid all terminators, and any partial matches on HTML tags.

Using `+?` makes the match non-greedy, which means it will stop as soon as it finds the first opening `\1`.

### CODE REVIEW

**Comment by antfu** [comment]
image

It seems wrong... I will take care of this later.

**Comment by ElMassimo** [comment]
@antfu Have in mind that we are doing two passes, later with `regexClassCheck`, which will not match any of the ones you highlighted.

File: src/index.ts [hunk 1]
```diff
@@ -119,7 +119,7 @@ function VitePluginWindicss(options: UserOptions = {}): Plugin[] {
     debug.detect(id)
     // classes
     Array.from(code.matchAll(regexQuotedString))
-      .flatMap(m => m[2]?.split(' ') || [])
+      .flatMap(m => m[2]?.split(/[\s'"`{}]/g) || [])
```
Comment (antfu) [inline src/index.ts:122]: Something the fix could be much simpler 😉 @ElMassimo

**Comment by antfu** [comment]
It's fixed now.

**Comment by antfu** [comment]
> @antfu Have in mind that we are doing two passes, later with `regexClassCheck`, which will not match any of the ones you highlighted.

That's not the problem. The problem is the string capture is wrong, some of the strings are not counted.

**Pull request merged**